1_Installation_Files (v1.5.5)/YOUR_ADMIN/image_handler.php

```diff
@@ -27,12 +27,12 @@
 define('HEADING_TITLE', IH_HEADING_TITLE);
 define('HEADING_TITLE_PRODUCT_SELECT', IH_HEADING_TITLE_PRODUCT_SELECT);
 
-$ih_page = isset($_GET['ih_page']) ? $_GET['ih_page'] : 'manager';
+$ih_page = $_GET['ih_page'] ?? 'manager';
```

Comment:
While I'm a fan of using the null-coalesce `??` syntax, it's only valid since PHP 7.0
https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op

Reply:
Evidently I am mainly guided by the IDE being set to 8: not to use new bells and whistles like this, but to point out impending deprecations. But it will suggest these things. Still, do we really have to support pre-php7?

Reply:
> do we really have to support pre-php7? 

I hear you. It's the perpetual challenge with open source code.

Here's the guiding principle: Are you willing to be on call to help the person who's installing the latest version of this plugin on their cart but has an older PHP version and can't upgrade cuz it'll break their cart and they're just trying to get some sales so they can justify the cost/time investment in upgrading?

Reply:
I'm sure Cindy would like to eliminate having to support older PHP versions for all plugins, but (for plugins especially) it's definitely less painful just to code to the lowest-common-sensible-denominator and avoid having to hand-hold those without the technical know-how for troubleshooting/fixing their own messes.

Reply:
I think that just the fact that ZC exists is already giving any startup a gift horse. But if someone wants to have a business, they should play their part and be business-like: if they do not have the skills to maintain the code, they should pay someone who does. Taking peoples money and handling their data is not a hobby.
 I think that the workload required to keep the code compliant with the rapid changes in php, improve existing code/fix bugs and add functionality is already far more than any non-contributing user can demand. Adding support for obsolete php and the absolute novice is well beyond any call of duty. They have to step up, not the project to bend down.
In my opinion.

Maybe plugins should also be marked with the php version they were tested on, as well as their ZC version.